packages/core/src/telemetry/service.ts

```diff
@@ -0,0 +1,165 @@
+import { exec } from "child_process";
+import Conf from "conf";
+import { randomBytes } from "crypto";
+import { createHash } from "node:crypto";
+import pc from "picocolors";
+
+import { type Options } from "@/config/options";
+import { getAnonymousMeta } from "@/telemetry/anonymous-meta";
+
+type TelemetryEvent = {
+  eventName: string;
+  payload: object;
+};
+
+type TelemetryConfig = {
+  enabled: boolean;
+  notifiedAt: string;
+  anonymousId: string;
+  salt: string;
+};
+
+async function _getProjectIdByGit() {
+  try {
+    let resolve: (value: Buffer | string) => void, reject: (err: Error) => void;
+    const promise = new Promise<Buffer | string>((res, rej) => {
+      resolve = res;
+      reject = rej;
+    });
+
+    exec(
+      `git config --local --get remote.origin.url`,
+      {
+        timeout: 1000,
+        windowsHide: true,
+      },
+      (error: null | Error, stdout: Buffer | string) => {
+        if (error) {
+          reject(error);
+          return;
+        }
+        resolve(stdout);
+      }
+    );
+
+    return String(await promise).trim();
+  } catch (_) {
+    return null;
+  }
+}
+
+async function getRawProjectId(): Promise<string> {
+  return (await _getProjectIdByGit()) ?? process.cwd();
+}
+
+export class TelemetryService {
+  private readonly conf: Conf<TelemetryConfig>;
+  private readonly sessionId: string;
+  private readonly TELEMETRY_DISABLED: boolean;
+  private distDir: string;
+  private loadProjectId: string | null = null;
+
+  constructor(options: Options) {
+    this.conf = new Conf({ projectName: "ponder", cwd: options.ponderDir });
+    this.distDir = options.ponderDir;
+    this.TELEMETRY_DISABLED = Boolean(process.env.TELEMETRY_DISABLED);
```

Comment:
I would do some manual testing do make sure this works as intended (if you haven't already). Also might just call the instance property `this.disabled`.